### DISCUSSION TITLE
Using AnyCodable instead of Any

### DISCUSSION THREAD

**Discussion by cbaker6** [netreconlab/Parse-Swift#56] [General]
There have been questions around Any in the past. As the compiler error states, Any isn’t Codable. The correct way to type erase a Codable typed is here: https://stackoverflow.com/a/71851227/4639041

**Comment by cbaker6** [comment]
@project-academy if you have questions about ParseSwift, I recommend you ask them here and use this repo instead as I don’t add features to the parse-community repo anymore, more details here: https://github.com/netreconlab/Parse-Swift/discussions/7

your original question has been answered above.

**Reply by proj-sashido** [reply]
Thanks! Apologies as I tried but didn’t see anything on SO

**Reply by cbaker6** [reply]
I recommend looking at playgrounds for examples on how to use Config and the rest of the SDK https://github.com/netreconlab/Parse-Swift/blob/main/ParseSwift.playground/Pages/14%20-%20Config.xcplaygroundpage/Contents.swift

**Reply by proj-sashido** [reply]
Yep! That’s where my confusion came from 😅
It had info for String and Int, but not for JSON/Dictionary/‘Object'

My project is now using this repo!

Would be interesting to put in examples for AnyCodable into the SDK. As well as mentioning that `config.save()` will cause an exception unless the master key has been supplied.

**Reply by cbaker6** [reply]
The example above should help and `AnyCodable` has [a number of examples on their site](https://flight.school/books/codable/). I also provide ParseCloud examples with AnyCodable here https://community.parseplatform.org/t/how-to-return-from-parse-cloud-function-in-node-js/2303/11?u=cbaker6

**Reply by cbaker6** [reply]
> Would be interesting to put in examples for AnyCodable into the SDK.

The SDK is built with zero dependencies meaning the Playgrounds has zero dependencies so examples using another framework won’t work there. AnyCodable is simply a type itself used for type erasing, so it works similarly as fundamental types like Int, String, Dictionary, etc. In addition, AnyCodable is just one framework that provides type erasure, there are many out there and it’s impossible to show examples for all. 

> As well as mentioning that `config.save()` will cause an exception unless the master key has been supplied.

A note can be added to the API docs, but the Parse Server docs for `Config` state that the `masterKey` is needed for updates. I would expect that if a developer is trying to use `Config` they will be familiar with the server docs: https://docs.parseplatform.org/rest/guide/#config. In most cases the `Config` shouldn't be updated from a client app and instead only be updated from Cloud Code by using [ParseServerSwift](https://github.com/netreconlab/parse-server-swift) or Cloud Code JS.

**Comment by proj-sashido** [comment]
_Edit: I answered my own question!_
Just saw [this](https://github.com/parse-community/Parse-Swift/pull/127#issue-866870311l) which states:
> _If developers want to use AnyCodable …  they should create them or add AnyCodable manually using SPM_

---

_(Disregard below)_

> Do I need to add AnyCodable as a dependency to my own project? 
> I can see it is being used inside the ParseSwift SDK but if I try to use it in my own project, I’m getting `"Cannot find type 'AnyCodable' in scope”`
> 
> When I add the Flight-School/AnyCodable repo as dependency, then it works fine, but I just wanted to make sure that it’s intentional that I can’t access it through ParseSwift SDK

**Reply by cbaker6** [reply]
This should help answer this https://github.com/parse-community/Parse-Swift/issues/141
